### PR TITLE
Fix quit of the app

### PR DESCRIPTION
**Pull request by petrvecera**
https://www.electronjs.org/docs/latest/api/app
https://stackoverflow.com/questions/52105917/electron-windows-app-close-not-ending-threads

### CODE REVIEW

**Review by JohannesMerkt — COMMENTED** [review]
while this fixes the app not terminating. It is not how the app was intended to work. Now you can only stop the app by closing all windows. previously you only had to close the main window to terminate the app. also the run in tray mode does not work anymore, since the app quits when all windows are closed. the intension of the run in tray mode was that you dont have to have a window open and can run the app in the "background". However this is a good hint on how to fix this issue. I will have a look how we can use this solution to retain the intended behaviour

**Comment by JohannesMerkt** [comment]
I found the reason:

I used destroy when closing a window. The documentation says: destroy skips the on window closed callback. The close callback was responsible for quiting the app but never was called.

File: packages/app/src/main/applicationManager.ts [hunk 1]
```diff
@@ -236,6 +237,10 @@ export class ApplicationManager {
 
   quit = (): void => {
     this.isQuitting = true;
+    this.windows.main.destroy();
+    this.windows.web.destroy();
+    this.windows.settings.destroy();
+    this.windows.about.destroy();
     app.quit();
```
Comment (petrvecera) [inline packages/app/src/main/applicationManager.ts:244]: What about command exit? https://www.electronjs.org/docs/latest/api/app#appexitexitcode 
I think the app.quit is "soft" quit with events and such. The exit should 100% kill everything?

File: packages/app/src/main/applicationManager.ts [hunk 2]
```diff
@@ -207,6 +207,7 @@ export class ApplicationManager {
       } else {
         this.inTrayMode = false;
         this.windows.main.show();
+        this.windows.settings.show();
```
[the remark this answers is not in the document]
Comment (JohannesMerkt) [reply packages/app/src/main/applicationManager.ts:210]: The first show on main is needed to ensure when we leave the run in tray mode that the main window is open. The second one is to make the settings app focused again. Its a bit irritating when you change a setting and suddently a different window pops over your settings window

**Pull request merged**